app/src/main/java/edu/gatech/spacebarz/buzzshelter/model/FirebaseDBManager.java

```diff
@@ -58,16 +57,14 @@ public static Shelter retrieveShelterInfo(String uid) throws DatabaseException {
     }
 
     public static void updateShelterInfo(Shelter updated) throws DatabaseException {
-        StoreObjectSynchronousTask<Shelter> task = new StoreObjectSynchronousTask<>(DatabaseKey.SHELTER, updated.getKey());
+        StoreObjectSynchronousTask<Shelter> task = new StoreObjectSynchronousTask<>(DatabaseKey.SHELTER, updated.getUID());
         DatabaseException ex = task.run(updated);
         if (ex != null) {
             throw ex;
         }
     }
 
     public static void insertNewShelterInfo(Shelter shelter) throws DatabaseException {
-        String uid = generateUID(DatabaseKey.SHELTER);
```

Comment:
Is there a reason that this block of code was removed? This will end up failing since the shelter being inserted won’t have a UID

Reply:
The UID for the shelters is obtained from the CSV. We do not need to generate our own.

Reply:
Right, but admins need a way to add new shelters to the database, so those UIDs wouldn't come from the CSV

Reply:
Additionally this means insertNewShelterInfo and updateShelter are now identical which can be confusing if in fact admins shouldn't be able to make new shelters

Reply:
We should have the same id scheme across the whole system. Want to use this UID generation code in the CSV uploader and put the CSV id in a separate field?

Reply:
I don't think we should have two sets of UIDs because they are only used for identification purposes. If we are going to use a standard UID scheme then we will have to ignore the CSV UIDs and instead use `push()` in the CSV uploader. I think having a UID and a CSV UID is confusing since the CSV UID would never be used

Reply:
I wasn't sure if they would want us to store the CSV UID for the course.

We could have a reserved block of high number UIDs set aside for mobile uploads, then have a db entry that tracks the next available id. Not practical, but preserves CSV id scheme. 

Push seems like the better way to go.

Reply:
Since we don't need to implement that method until a later milestone we can just ask a TA to see if we need to preserve the CSV UIDs